### PR TITLE
Added Nvidia-Dali Builder and Resolver

### PR DESCRIPTION
**Pull request by chaitravi-ce**
Signed-off-by: chaitravi-ce <chaitravi007@gmail.com>

What do these changes do?
-------------------------

* Added Builder and Resolver for Nvidia-Dali
* Added Tests

Related issue number
--------------------

Related to #280

### CODE REVIEW

File: setup_ml.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def find_ml_packages(root):
         "install": install_plat
     },
     zip_safe=False,
-    install_requires=['vineyard', 'tensorflow', 'xgboost', 'torch', 'mxnet'],
+    install_requires=['vineyard', 'tensorflow', 'xgboost', 'torch', 'mxnet', 'nvidia-dali-cuda110'],
```
Comment (sighingnow) [inline setup_ml.py:84]: I think we need to add a profile, namely `gpu`, where the `nvidia-dali-cuda110` is listed as a dependency, as it is not available on GPU.

File: python/vineyard/contrib/ml/tests/test_dali.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            register_dali_types(builder, resolver)
+            yield builder, resolver
+
+
+num_gpus = 1
+device_id = 0
+batch_size = 2
+num_threads = 4
+
+
```
Comment (sighingnow) [inline python/vineyard/contrib/ml/tests/test_dali.py:44]: Skip those tests if the dali package is not available.

File: setup_ml.py (hunk 1 above)
Comment (chaitravi-ce) [reply setup_ml.py:84]: Yes dali requires cuda and nvidia drivers. I am not sure how we can add it here

Comment (sighingnow) [reply setup_ml.py:84]: Its all right, I can assistant to revise the PR.

Comment (chaitravi-ce) [reply setup_ml.py:84]: Thank You

**Comment by sighingnow** [comment]
@chaitravi-ce I have added guard to check if dali is applicable in https://github.com/v6d-io/v6d/pull/528/commits/700da7fcea06d2ecc306635692b99028c33c4ad1, we can get this PR merged once CI turns green.

Thanks!

**Pull request merged**